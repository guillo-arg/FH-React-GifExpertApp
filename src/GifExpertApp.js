import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setcategories] = useState(defaultCategories);

    const handleAdd = () =>{
        setcategories([...categories, 'Pokemon']);
    };
  return (
      <>
        <h2>GifExpertApp</h2>
        <hr />
        <AddCategory setCategories = {setcategories}/>

        <ol>
            {categories.map( category =>{
                return <GifGrid 
                            category={category}
                            key={category}
                        />
            })}
        </ol>
      </>
  );
};
