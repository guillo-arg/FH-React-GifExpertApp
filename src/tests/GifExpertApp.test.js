import { GifExpertApp } from "../GifExpertApp";
import {shallow} from 'enzyme';
;

describe('Pruebas sobre <GifExpertApp />', () => {
  
    test('Debe cargar el componente correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['goku', 'vegeta'];
        const wrapper = shallow(< GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});
