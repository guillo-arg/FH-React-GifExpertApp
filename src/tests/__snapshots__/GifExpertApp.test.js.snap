// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExpertApp /> Debe cargar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="goku"
      key="goku"
    />
    <GifGrid
      category="vegeta"
      key="vegeta"
    />
  </ol>
</Fragment>
`;
